packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -91,6 +105,28 @@ export default declare(
                   importPath.remove();
                 }
               }
+
+              continue;
+            }
+
+            if (stmt.isExportDeclaration()) {
+              stmt = stmt.get("declaration");
+            }
+
+            if (stmt.isVariableDeclaration({ declare: true })) {
+              for (const name of Object.keys(stmt.getBindingIdentifiers())) {
+                registerGlobalType(path.scope, name);
+              }
+            } else if (
+              stmt.isTSTypeAliasDeclaration() ||
+              stmt.isTSDeclareFunction() ||
```

Comment:
Would `stmd.node.id.name` bail on `export default function() {}`?

Reply:
`TSDeclareFunction` (`declare function f(): type;`) can't be used in an `export default` declaration, so it can never be anonymous.